src/function/FunctionRestriction.ts

```diff
@@ -1,63 +1,47 @@
-import {Restrictions} from "../config/Restrictions";
-import {ProjectElements, ProjectLinks, VocabularyElements} from "../config/Variables";
-import {RestrictionObject} from "../datatypes/RestrictionObject";
-import {parsePrefix} from "./FunctionEditVars";
-import {ConnectionObject} from "../datatypes/ConnectionObject";
+import {ProjectElements, ProjectLinks, ProjectSettings, VocabularyElements} from "../config/Variables";
+import {Restriction} from "../datatypes/Restriction";
+import {getNewLink} from "./FunctionGetVars";
+import {LinkType} from "../config/Enum";
+import {addLink} from "./FunctionCreateVars";
 
-export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }) {
-	if (target.type !== "bnode" && (restriction in Restrictions)) {
-		let newRestriction = new RestrictionObject(restriction, onProperty, target.value);
+export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }, onClass?: string) {
+	if (target.type !== "bnode") {
+		let newRestriction = new Restriction(restriction, onProperty, target.value, onClass);
 		for (let rest of obj[iri].restrictions) {
 			if (rest.target === newRestriction.target
 				&& rest.restriction === newRestriction.restriction
-				&& rest.onProperty === newRestriction.onProperty) {
+				&& rest.onProperty === newRestriction.onProperty
```

Comment:
It would be probably worth building a dedicated comparator function here - something like `Restriction.isSame(other: Restriction)`.

Reply:
Implemented. Thank you for the comment.